resources/testharness.js

```diff
@@ -1056,6 +1056,37 @@ policies and contribution forms [3].
     }
     expose(assert_array_equals, "assert_array_equals");
 
+    function assert_array_members_are_same(actual, expected, description)
+    {
+      assert(typeof actual === "object" && actual !== null && "length" in actual
+             && "sort" in actual && "sort" in expected,
+             "assert_array_members_are_same", description,
+             "value is ${actual}, expected array",
+             {actual:actual});
+      assert(actual.length === expected.length,
+             "assert_array_members_are_same", description,
+             "lengths differ, expected ${expected} got ${actual}",
+             {expected:expected.length, actual:actual.length});
+
+      for (var i = 0; i < actual.length; i++) {
```

Comment:
This loop is marginally surprising to me given what this assertion otherwise does. Do we really want to check that both arrays have holes at the same indices? Arguably what we want to check is that they have the same number of holes, no? (Though really once you get into treating arrays as sets, it's very unclear what holes mean!)

Reply:
So generally speaking no and to be honest I was assuming the input to this API is always arrays. 

>  Do we really want to check that both arrays have holes at the same indices? Arguably what we want to check is that they have the same number of holes, no?
This is what is currently been done for `assert_arrays_equal`. Essentially, in the example
```
x = [1, 2, ,3]
y = [1,2,undefined,3]
z = (x[2] === y[2]) // evaluates to true since both return undefined.
```
but,`x.hasOwnProperty(2) !== y.hasOwnProperty(2)` . So technically here we are weeding out some of the js-arrays which do not have integer contiguous indices.